app/assets/javascripts/common/toggleScrollToPosition.js

```diff
@@ -0,0 +1,31 @@
+// Check for scroll position and scroll
+window.addEventListener("DOMContentLoaded", (event) => {
+  if (doesYPositionExistInLocalStorage()) {
+    scrollToYPosition();
+    removeYPositionLocalStorage();
+  }
+});
+
+// Create toggle change listener and add y position on 'change' event
+function createToggleListener(id) {
+  const toggleElement = document.getElementById(id);
+  toggleElement.addEventListener("change", (event) => {
+    setYPositionLocalStorage();
+  });
+}
```

Comment:
I've re-written the whole thing to be a generic function that creates a listener for each toggle. All that is required is the toggle ID value

Reply:
[suggestion, non-blocking] Perhaps this could be renamed to something like `createScrollTrackingListener`. Essentially, something that would convey what event we are interested in listening to.

Reply:
Done